docs/pages/Cardinal/API-Reference/World.mdx

```diff
@@ -0,0 +1,529 @@
+
+```go
+func WithAdapter(adapter shard.Adapter) Option
+```
+
+| Parameter | Type              | Description                              |
+|-----------|-------------------|------------------------------------------|
+| adapter   | shard.Adapter     | The adapter to be associated with World. |
+
+#### WithReceiptHistorySize
+
+The `WithReceiptHistorySize` option informs the `World` object on how many ticks the `World` should keep receipts for. For example, if we are on tick 40, and have a receipt history size of 5, the `World` will have receipts for ticks 35-39. On tick 41, the `World` will have receipts for ticks 36-40.
+
+```go
+func WithReceiptHistorySize(size int) Option
+```
+
+| Parameter | Type     | Description                                           |
+|-----------|----------|-------------------------------------------------------|
+| size      | int      | The size of the receipt history to be set for World. |
+
+
+#### WithNamespace
+
+The `WithNamespace` allows the `World` to have a custom Namespace. Namespaces are used to differentiate `World` objects that post data to the EVM base shard.
+
+```go
+func WithNamespace(id string) Option
+```
+| Parameter | Type    | Description                                   |
+|-----------|---------|-----------------------------------------------|
+| ns        | string  | The namespace to be set for World. |
+
+
+## Methods
+
+### AddSystems
+
+`AddSystems` adds one or more systems to the `World`. Systems are executed in the order of which they were added to the world.
+
+```go
+func (w *World) AddSystems(s ...System)
+```
+
+#### Example
+
+```go
+package main
+
+import (
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/systems"
+)
+
+func main() {
+    // ... world setup ...
+
+    world.AddSystems(systems.MoveSystem, systems.HealthRegenSystem, systems.AttackSystem)
+}
+```
+
+#### Parameters
+
+| Parameter | Type         | Description                                             |
+|-----------|--------------|---------------------------------------------------------|
+| s         | ...System    | Variadic parameter for systems to be added to the World. |
+
+
+### RegisterComponents
+
+`RegisterComponents` registers one or more components to the `World`. Upon registration, components are assigned an ID. IDs are assigned incrementally, starting from 0, in the order in which they were passed to the method.
+
+
+<Callout type={"warning"}>
+    `RegisterComponents` can be only be called once. Subsequent calls to the method will return an error.
+</Callout>
+
+
+```go
+func (w *World) RegisterComponents(components ...component.IComponentType) error
+```
+#### Example
+
+```go
+package main
+
+import (
+	"log"
+
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/components"
+)
+
+func main() {
+    // ... world setup ...
+
+    err := world.RegisterComponents(
+		components.LocationComponent,
```

Comment:
package name in starter-game-template is singular `component` would like to keep it consistent

Reply:
[package consistency:](https://github.com/Argus-Labs/world-engine/pull/200/commits/3ace678c1ac3dfe79a376c766e89708b80e22404)